huaweicloud/services/dns/resource_huaweicloud_dns_endpoint.go

```diff
@@ -0,0 +1,324 @@
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        5 * time.Second,
+		PollInterval: 5 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf(
+			"error waiting for DNS endpoint (%s) to become ACTIVE for creation: %s",
+			endpoint.ID, err)
+	}
+
+	return resourceDNSEndpointRead(ctx, d, meta)
+}
+
+func waitForDNSEndpoint(client *golangsdk.ServiceClient, id string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		endpoint, err := endpoints.Get(client, id).Extract()
+		if err != nil {
+			if _, ok := err.(golangsdk.ErrDefault404); ok {
+				return endpoint, "DELETED", nil
+			}
+			return nil, "", err
+		}
+
+		log.Printf("[DEBUG] DNS endpoint (%s) current status: %s", endpoint.ID, endpoint.Status)
+		return endpoint, parseStatus(endpoint.Status), nil
+	}
+}
+
+func resourceDNSEndpointUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	dnsClient, err := conf.DNSV21Client(region)
+	if err != nil {
+		return diag.Errorf("error creating DNS client: %s", err)
+	}
+
+	name := d.Get("name").(string)
+
+	if d.HasChange("name") {
+		opts := endpoints.UpdateOpts{Name: name}
+		_, err = endpoints.Update(dnsClient, d.Id(), opts).Extract()
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	log.Printf("[DEBUG] Waiting for DNS endpoint (%s) to become available", d.Id())
+	stateConf := &resource.StateChangeConf{
+		Target:       []string{"ACTIVE"},
+		Pending:      []string{"PENDING"},
+		Refresh:      waitForDNSEndpoint(dnsClient, d.Id()),
+		Timeout:      d.Timeout(schema.TimeoutUpdate),
+		Delay:        5 * time.Second,
+		PollInterval: 5 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf(
+			"error waiting for DNS endpoint (%s) to become ACTIVE for update: %s",
+			d.Id(), err)
+	}
+
+	return resourceDNSEndpointRead(ctx, d, meta)
+}
+
+func resourceDNSEndpointRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	dnsClient, err := conf.DNSV21Client(region)
+	if err != nil {
+		return diag.Errorf("error creating DNS client: %s", err)
+	}
+
+	id := d.Id()
+	endpointInfo, err := endpoints.Get(dnsClient, id).Extract()
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving DNS endpoint")
+	}
+	log.Printf("[DEBUG] Retrieved endpoint %s: %#v", id, endpointInfo)
+
+	ipAddress, err := ipaddress.List(dnsClient, id).Extract()
+	if err != nil {
+		return diag.Errorf("error retrieving ip addresses: %s", err)
+	}
+	log.Printf("[DEBUG] Retrieved ip address %s: %#v", id, ipAddress)
+
+	subnetClient, err := conf.NetworkingV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating VPC client: %s", err)
+	}
+	subnetList, err := subnets.List(subnetClient, subnets.ListOpts{})
```

Comment:
how about filter the subnets with vpc_id?

Reply:
only subnet_id, unsuppot to search

Reply:
got it, updated